### PR TITLE
Fix notebooks on CI - Try to speed them up a little bit

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR works around #1005
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [ ] HISTORY.rst has been updated (with summary of main changes)
  - [ ] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.
- [ ] `bumpversion patch` has been called on this branch
- [ ] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?

* Pin IPython to older than 8.0. See issue for details.
* Reactivate the nbstripout hook.
* Minor changes to the sdba and ensemble-advanced notebooks. I think we gained 2-3 minutes, not much but something.

### Does this PR introduce a breaking change?
No. But I did like Ipython 8, I'm a bit sad.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
-          python-version: "3.8"
```
Comment (Zeitsperre) [inline .github/workflows/main.yml:41]: Why not Python3.8 here?

File: tox.ini [hunk 2]
```diff
@@ -29,7 +29,7 @@ whitelist_externals =
 commands =
     - mypy xclim
     pylint --rcfile=setup.cfg --exit-zero xclim
-    pytest --nbval docs/notebooks --durations=3
+    pytest -x --nbval docs/notebooks --durations=3
```
Comment (Zeitsperre) [inline tox.ini:32]: Do we still want to abort on first failure?

**Review by Zeitsperre — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (aulemahal) [reply .github/workflows/main.yml:41]: woups I haven't pushed my other revert commit. We should use python 3.8 indeed.

File: tox.ini (hunk 2 above)
Comment (aulemahal) [reply tox.ini:32]: Same as above. I don't think we want to abort on first failure.

**Pull request merged**